qemu/tests/block_resize.py

```diff
@@ -119,130 +119,143 @@ def _check_event_cmd(cmd):
+                session = vm.reboot(session=session)
+            if params.get("need_rescan") == "yes":
+                drive.rescan_disks(session)
+
+            # We need extend disk in monitor first then extend it in guest
+            if float(ratio) > 1.0:
+                error_context.context("Extend disk to %s in guest"
+                                      % block_size, test.log.info)
+                if os_type == 'windows':
+                    max_block_size = int(params["max_block_size"])
+                    if int(block_size) >= max_block_size:
+                        test.cancel(
+                            "Cancel the test for more than maximum %dB disk." %
+                            max_block_size)
+                    drive.extend_volume(session, mpoint)
+                else:
+                    utils_disk.resize_partition_linux(session, partition,
+                                                      str(block_size))
+                    utils_disk.resize_filesystem_linux(session, partition,
+                                                       utils_disk.SIZE_AVAILABLE)
+            global current_size
+            current_size = 0
+            steps = params.get_numeric("verify_disk_size_steps", 1)
+            if not wait.wait_for(lambda: verify_disk_size(session, os_type,
+                                                          disk), 20, 0, steps,
+                                 "Block Resizing"):
+                test.fail("Block size get from guest is not same as expected.\n"
+                          "Reported: %s\nExpect: %s\n" % (current_size, block_size))
+            session = vm.reboot(session=session)
+
+            if os_type == 'linux':
+                # After guest reboot, reget the disk letter, if it changed, replace
+                # variables, i.e 'mpoint', 'partition', 'disk' and 'md5_filename'
+                new_disk = utils_misc.get_linux_drive_path(session, disk_serial)
+                new_disk = re.search(r"([svh]d\w+)", new_disk, re.M).group(0)
+                if new_disk != disk:
+                    mpoint = mpoint.replace(disk, new_disk)
+                    partition = partition.replace(disk, new_disk)
+                    disk = new_disk
+                    if params.get('md5_test') == 'yes':
+                        md5_filename = mpoint + junction + md5_file
+                    session.cmd("mkdir -p %s" % mpoint)
+
+                if not utils_disk.is_mount(partition, dst=mpoint,
+                                           fstype=fstype, session=session):
+                    res = utils_disk.mount(partition, mpoint,
+                                           fstype=fstype, session=session)
+                    if not res:
+                        test.fail("Mounting data disk was failed! ")
+
+            if params.get('iozone_test') == 'yes':
+                iozone_timeout = float(params.get("iozone_timeout", 1800))
```

Comment:
The following method also work well. All look good to me.
iozone_timeout = params.get_numeric("iozone_timeout", 1800, float)